File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    idlen = len(id_list)
    
    report = set(report) 
    
    idlist = [[0] * idlen for _ in range(idlen)] 
    iddict = {id_list[i]: i for i in range(idlen)}
    
    for i in report:
        a, b = i.split()
        idlist[iddict[a]][iddict[b]] = 1 
    
    rlist = [0] * idlen
    for i in range(idlen):
        for j in range(idlen):
            if idlist[j][i] == 1:
                rlist[i] += 1
    
    answer = []
    for i in range(idlen):
        temp = 0
        for j in range(idlen):
            if idlist[i][j] == 1 and rlist[j] >= k:
                temp += 1
        answer.append(temp)
        
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

int compare_strings(const void* a, const void* b) {
    return strcmp(*(const char**)a, *(const char**)b);
}

int get_user_index(const char* name, const char* id_list[], int n) {
    for (int i = 0; i < n; i++) {
        if (strcmp(name, id_list[i]) == 0) {
            return i;
        }
    }
    return -1;
}

int* solution(const char* id_list[], size_t id_list_len, const char* report[], size_t report_len, int k) {
    int n = (int)id_list_len;
    
    int* answer = (int*)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++) answer[i] = 0;
    
    qsort(report, report_len, sizeof(char*), compare_strings);
    
    int* report_count = (int*)calloc(n, sizeof(int));
    
    for (size_t i = 0; i < report_len; i++) {
        if (i > 0 && strcmp(report[i], report[i - 1]) == 0) {
            continue;
        }
        
        char buf[50];
        strcpy(buf, report[i]);
        char* reporter = strtok(buf, " ");
        char* reported = strtok(NULL, " ");
        
        int reported_idx = get_user_index(reported, id_list, n);
        report_count[reported_idx]++;
    }
    
    for (size_t i = 0; i < report_len; i++) {
        if (i > 0 && strcmp(report[i], report[i - 1]) == 0) {
            continue;
        }
        
        char buf[50];
        strcpy(buf, report[i]);
        char* reporter = strtok(buf, " ");
        char* reported = strtok(NULL, " ");
        
        int reporter_idx = get_user_index(reporter, id_list, n);
        int reported_idx = get_user_index(reported, id_list, n);
        
        if (report_count[reported_idx] >= k) {
            answer[reporter_idx]++;
        }
    }
    
    free(report_count);
    return answer;
}